images/alpine/cloudbuild.yaml

```diff
@@ -1,19 +1,14 @@
 steps:
-  - name: gcr.io/cloud-builders/docker
+  - name: gcr.io/k8s-testimages/gcb-docker-gcloud
+    entrypoint: /buildx-entrypoint
     args:
     - build
-    - -t
-    - gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
+    - --tag=gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
+    - --tag=gcr.io/$PROJECT_ID/alpine:latest
+    - --platform=linux/amd64,linux/arm64/v8
     - --build-arg=IMAGE_ARG=gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
+    - --push
     - .
     dir: .
-  - name: gcr.io/cloud-builders/docker
-    args:
-    - tag
-    - gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
-    - gcr.io/$PROJECT_ID/alpine:latest
 substitutions:
   _GIT_TAG: '12345'
-images:
-  - 'gcr.io/$PROJECT_ID/alpine:$_GIT_TAG'
```

Comment:
What is the reason for dropping these? (or rather moving pushing them into the build step)
https://cloud.google.com/build/docs/build-config#images

Reply:
This doesn't seem to work with manifests.
Using:
```
steps:
  - name: gcr.io/k8s-testimages/gcb-docker-gcloud
    entrypoint: /buildx-entrypoint
    args:
    - build
    - --tag=gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
    - --tag=gcr.io/$PROJECT_ID/alpine:latest
    - --platform=linux/amd64,linux/arm64/v8
    - --build-arg=IMAGE_ARG=gcr.io/$PROJECT_ID/alpine:$_GIT_TAG
    - .
    dir: .
substitutions:
  _GIT_TAG: '12345'
images:
  - 'gcr.io/$PROJECT_ID/alpine:$_GIT_TAG'
  - 'gcr.io/$PROJECT_ID/alpine:latest'

```
yields:
```
ERROR: failed to find one or more images after execution of build steps: ["gcr.io/k8s-staging-test-infra/alpine:v20210618-1bc5e9d8bf" "gcr.io/k8s-staging-test-infra/alpine:latest"]
```

Reply:
GCR specific but cheap and simple https://cloud.google.com/sdk/gcloud/reference/container/images/add-tag

To do it generically you'd need to push the manifest again as a tag (and only the manifest, the underlying resources should already be pushed), [`crane`](https://github.com/google/go-containerregistry/tree/main/cmd/crane) can do this (and other super useful things!) see: [`crane tag`](https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_tag.md)